### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by dlorenc**
Publish appender.par
  Add tool for flattening images

cc @nkubala

### CODE REVIEW

**Comment by nkubala** [comment]
thanks. what's up with travis?

File: client/v2_2/docker_image_.py [hunk 1]
```diff
@@ -120,6 +120,58 @@ def __str__(self):
...
+    return self._image.fs_layers()
+
+  def config_blob(self):
+    """Override."""
+    return self._image.config_blob()
+
+  def blob_set(self):
+    """Override."""
+    return self._image.blob_set()
+
```
Comment (KingEmet) [inline client/v2_2/docker_image_.py]:
```
def digest(self):
  """Override."""
  return self._image.digest()
```

File: client/v2_2/docker_image_.py [hunk 2]
```diff
@@ -352,12 +405,12 @@ def _gzipped_content(self, name):
   def _populate_manifest_and_blobs(self):
     """Populates self._manifest and self._blob_names."""
     # TODO(user): Update mimes here for oci_compat.
+    config_blob = 'sha256:' + hashlib.sha256(self.config_file()).hexdigest()
```
Comment (KingEmet) [inline client/v2_2/docker_image_.py]: config_blob = util.Digest(self.config_file())

Comment (KingEmet) [reply client/v2_2/docker_image_.py]: Might want genericize the hashing util to not refer to just manifests...? @mattmoor

**Comment by mattmoor** [comment]
@KingEmet This is just publishing what we have internally, so let's not hand edit it here.

@jonjohnsonjr Can you fix `oci_compat` in the mono-repo to unblock this?

**Comment by dlorenc** [comment]
> @jonjohnsonjr Can you fix oci_compat in the mono-repo to unblock this?

Grabbed the fix from the monorepo.

**Comment by dlorenc** [comment]
@jonjohnsonjr this looks good to go now, thanks for the fix!

**Pull request merged**